email_processing.py

```
@@ -0,0 +1,197 @@
+        date = '0001-01-01T00:00:00.000Z'
+        document_list = self.all_files
+        annotation_link = None
+
+        def convertDate(d):
+            new_date = datetime.datetime.strptime(d, "%Y-%m-%dT%H:%M:%S.%fZ")
+            return (new_date.date(), new_date.time())
+
+        date = convertDate(date)
+
+        print(f'Looking for latest uploaded file ...')
+        for i in document_list:
+            if i['original_file_name']==FILE_NAME:
+                if date < convertDate(i['arrived_at']):
+                    date = convertDate(i['arrived_at'])
+                    annotation_link = i['annotations']
+                    filename = i['original_file_name']
+        if annotation_link is None:
+            print('The file not found. It seems annotation link is empty.')
+            return 2
+
+        print(f'Latest uploaded file {FILE_NAME} and its annotation link: {annotation_link[0]}')
+        try:
+            r = requests.get(annotation_link[0], headers=headers)
+
+            if r.status_code in range(200, 299):
+                annotation_data = r.json()['status']
+                if annotation_data == 'to_review' and filename==FILE_NAME:
+                    print(f"The {filename} has been succesfully added to queue! OK.")
+                    return 0
+                elif annotation_data == 'failed_import':
+                    print(f"Import failed")
+                    return 1
+                else:
+                    print('It seems the file is still importing ... \nWaiting for 20 seconds and retrieving processing status again ...')
+                    time.sleep(20)
+                    self.check_processing()
+        except requests.exceptions.RequestException:
+                raise Exception(f'Failed to connect to {annotation_link[0]}') from None
+
+def main():
+    client = RossumAPI()
+    client.get_access_token()
+    client.import_file_by_email()
+    client.list_all_documents()
+    if client.check_processing() == 2:
```

I'd rather raise an exception than to check for return codes. For instance a book Clean code advises so. Moreover, Python is well constructed around exceptions and those are used often.